diff --git a/src/common/com-linux.cpp b/src/common/com-linux.cpp
index 710af6e3a..9fe312f42 100644
--- a/src/common/com-linux.cpp
+++ b/src/common/com-linux.cpp
@@ -464,7 +464,7 @@ int Comm::getDCD( void )
 
 void Comm::Flush( void )
 {
-    ioctl( m_fd, TCFLSH, 2);
+    tcflush( m_fd, TCIOFLUSH );
 }
 
 ///////////////////////////////////////////////////////////////////////////////
@@ -600,8 +600,12 @@ void Comm::RtsOff()
 
 int Comm::isTransmitterEmpty()
 {
+#ifdef TIOCSERGETLSR
     int rv;
     ioctl( m_fd, TIOCSERGETLSR, &rv );
     return rv;
+#else
+    return ( 0 == tcdrain( m_fd ) );
+#endif
 }
 
 